src/infra/aws/client/kms.go

```diff
@@ -116,25 +140,63 @@ func (c *AWSClient) Sign(_ context.Context, keyID string, msg []byte, signingAlg
 	return out, nil
 }
 
-func (c *AWSClient) DeleteKey(_ context.Context, keyID string) (*kms.ScheduleKeyDeletionOutput, error) {
+func (c *AWSClient) DeleteKey(ctx context.Context, keyID string) (*kms.ScheduleKeyDeletionOutput, error) {
 	out, err := c.kmsClient.ScheduleKeyDeletion(&kms.ScheduleKeyDeletionInput{
 		KeyId: &keyID,
 	})
 	if err != nil {
 		return nil, parseKmsErrorResponse(err)
 	}
 
+	err = c.waitKeyState(ctx, keyID, func(metadata *kms.KeyMetadata) error {
+		if !*metadata.Enabled && *metadata.KeyState == KeyStatePendingDeletion {
+			return nil
+		}
+
+		return fmt.Errorf("key %s was not scheduled for deletion yet", keyID)
+	})
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Here according to the state table, the operation will fail if the state is KeyPendingDeletion or Updating.
And in the case KeyPendingDeletion that should mean that deletion has already been scheduled

Reply:
the operation will fail in meanwhile the key is still enabled and not scheduled for deletion. I am not sure I understand what you are proposing, would u mind to rephrase the concern?

Reply:
Ok I know where my mistake is, I missed the fact that you were calling that check after invoking the ScheduleForDeletion, so here is ok. 
But should not it be called before ? 
The errors we have are on that call being rejected, because key has already been scheduled for deletion